src/napoleon_navigation_rosnode.cpp

```diff
@@ -618,77 +553,884 @@ int main(int argc, char** argv)
                 relative_angle = wrapToPi(obj3wall_angle-obj2frontwall_angle);
-    
+
                 if (OBJ2TASK[5].compare("right") == 0 && relative_angle < -SHARP_ANGLE_TRESHOLD) {
                     // Sharp angle to the right, we need to take the next wall into account as well
                     sharp_corner[ka+1] = true;
                 } else if (OBJ2TASK[5].compare("left") == 0 && relative_angle > SHARP_ANGLE_TRESHOLD) {
                     // Sharp angle to the left, we need to take the next wall into account as well
                     sharp_corner[ka+1] = true;
                 }
-            }            
+            }
         }
 
-    }    
-
+    }
+}
 
-    double v_ax = 0, theta_dot = 0, v_des, phi, v_scale;
-    bool ropod_is_in_44, ropod_is_in_45, ropod_is_in_46, ropod_is_in_47,
-         ropod_is_in_48, ropod_is_in_49, ropod_is_in_50, ropod_is_in_51;
 
-    bool consider_overtaking_area1ID, consider_overtaking_area3ID;
-    bool update_assignment;
-    int uprev;
-    bool ropod_colliding_obs = true;
-    bool ropod_colliding_wall = true;
 
-    AreaQuad current_entry = generateEntry(assignment[0], assignment[1], ENTRY_LENGTH, arealist, pointlist);
+void considerOvertaking()
+{
+    // Consider overtaking.
+    // Cesar -> TODO: Because of the changes to support consecutive hallway areas, this part still needs to be adapted
+    if (u < ka_max-1 && no_obs > 0) {
+        if (update_assignment) {
+            current_hallway = getAreaByID(area1ID, arealist);
+            if(next_area.type == "hallway")
+            {
+                next_hallway = next_area;
+            }
+            else
+            {
+                next_hallway = getAreaByID(area3ID, arealist);
+            }
+        }
+        // TODO: The contain function should consider not only the center but all corners
+        if (current_hallway.contains(obs_center_global)) {
+            consider_overtaking_area1ID = true;
+            // This strategy only allows to overtake 1 obstacle
+        }
+        if (next_hallway.contains(obs_center_global)) {
+            consider_overtaking_area3ID = true;
+            // This strategy only allows to overtake 1 obstacle
+        }
+    }
+}
 
-    std::ofstream myfile;
-    //myfile.open ("/simdata/ropod_" + get_date() +".txt");
-    myfile.open ("/home/cesar/Documents/simdata/ropod_" + get_date() +".txt");
-    myfile << "time" << "\t" << "tictoc" << "\t" << "state" << "\t" << "task counter" << "\t" << "tube width" << "\t" <<
-            "phi des" << "\t" << "v ropod odom" << "\t"<< "v ropod cmd" << "\t" << "x ropod" << "\t" << "y ropod" << "\t" <<
-            "theta ropod" << "\t" "x obs" << "\t" << "y obs" << "\t" << "theta obs" << "\t" << "obs width" << "\t" <<
-            "obs depth" << "\t" << "obs vx" << "\t" << "obs vy" << "\t" << "des accel" <<"\n";
 
-    std::clock_t start_loop;
+void overtakeStateMachine()
```

Comment:
It's better to move all these functions to separate ROS independent class.

Reply:
Indeed, this is just a first step. Due to the limited amount of time I have to spend on this task, first I first divided the main in these functions to improve the readability, which eventually would help continuing the refactoring by someone else. As agreed on previous discussions, further refactoring is needed.